Remove last_modified from sdoc template

zzak opened the pull request:
Copies ead9de6 for the sdoc template.
Note, this template was removed in 3000a97 from main (unstable 3.x branch).

Fixes #380

zzak committed:
Remove last_modified from sdoc template

Copies ead9de6 for the sdoc template.
Note, this template was removed in 3000a97 from main (unstable 3.x branch).

Kerrick commented:
FWIW I no longer get `undefined method 'last_modified'` with this branch, but I do get `undefined method 'each_constant'`.

```
kerrick@laptop-kerrick-primary sdoc_test % bundle exec sdoc .
Parsing sources...
100% [1431/1431]  tmp/cache/bootsnap/load-path-cache

Generating SDoc format into /Users/kerrick/Temporary/sdoc_test/doc...

You can visit the home page at: file:///Users/kerrick/Temporary/sdoc_test/doc/index.html
uh-oh! RDoc had a problem:
Error while evaluating /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/rdoc/generator/template/sdoc/_context.rhtml: undefined method 'each_constant' for an instance of RDoc::NormalClass (at "\n      <table border='0' cellpadding='5'>\n        ")

run with --debug for full backtrace
```

<details>
<summary>Full terminal log with `--debug` flag</summary>

```
kerrick@laptop-kerrick-primary ~ % rails -v
Rails 8.1.1
kerrick@laptop-kerrick-primary ~ % ruby -v 
ruby 3.4.7 (2025-10-08 revision 7a5688e2a2) +PRISM [arm64-darwin25]
kerrick@laptop-kerrick-primary ~ % cd ~/Temporary && rails new sdoc_test
      create  
      create  README.md
      create  Rakefile
      create  .ruby-version
      create  config.ru
      create  .gitignore
      create  .gitattributes
      create  Gemfile
         run  git init -b main from "."
Initialized empty Git repository in /Users/kerrick/Temporary/sdoc_test/.git/
      create  app
      create  app/assets/stylesheets/application.css
      create  app/controllers/application_controller.rb
      create  app/helpers/application_helper.rb
      create  app/jobs/application_job.rb
      create  app/mailers/application_mailer.rb
      create  app/models/application_record.rb
      create  app/views/layouts/application.html.erb
      create  app/views/layouts/mailer.html.erb
      create  app/views/layouts/mailer.text.erb
      create  app/views/pwa/manifest.json.erb
      create  app/views/pwa/service-worker.js
      create  app/assets/images
      create  app/assets/images/.keep
      create  app/controllers/concerns/.keep
      create  app/models/concerns/.keep
      create  bin
      create  bin/brakeman
      create  bin/bundler-audit
      create  bin/ci
      create  bin/dev
      create  bin/rails
      create  bin/rake
      create  bin/rubocop
      create  bin/setup
      create  bin/thrust
      create  Dockerfile
      create  .dockerignore
      create  bin/docker-entrypoint
      create  .rubocop.yml
      create  .github/workflows
      create  .github/workflows/ci.yml
      create  .github/dependabot.yml
      create  config
      create  config/routes.rb
      create  config/application.rb
      create  config/environment.rb
      create  config/bundler-audit.yml
      create  config/cable.yml
      create  config/ci.rb
      create  config/puma.rb
      create  config/storage.yml
      create  config/environments
      create  config/environments/development.rb
      create  config/environments/production.rb
      create  config/environments/test.rb
      create  config/initializers
      create  config/initializers/assets.rb
      create  config/initializers/content_security_policy.rb
      create  config/initializers/cors.rb
      create  config/initializers/filter_parameter_logging.rb
      create  config/initializers/inflections.rb
      create  config/initializers/new_framework_defaults_8_1.rb
      create  config/locales
      create  config/locales/en.yml
      create  config/master.key
      append  .gitignore
      create  config/boot.rb
      create  config/database.yml
      create  db
      create  db/seeds.rb
      create  lib
      create  lib/tasks
      create  lib/tasks/.keep
      create  log
      create  log/.keep
      create  public
      create  public/400.html
      create  public/404.html
      create  public/406-unsupported-browser.html
      create  public/422.html
      create  public/500.html
      create  public/icon.png
      create  public/icon.svg
      create  public/robots.txt
      create  script
      create  script/.keep
      create  tmp
      create  tmp/.keep
      create  tmp/pids
      create  tmp/pids/.keep
      create  vendor
      create  vendor/.keep
      create  test/fixtures/files
      create  test/fixtures/files/.keep
      create  test/controllers
      create  test/controllers/.keep
      create  test/mailers
      create  test/mailers/.keep
      create  test/models
      create  test/models/.keep
      create  test/helpers
      create  test/helpers/.keep
      create  test/integration
      create  test/integration/.keep
      create  test/test_helper.rb
      create  test/system
      create  test/system/.keep
      create  test/application_system_test_case.rb
      create  storage
      create  storage/.keep
      create  tmp/storage
      create  tmp/storage/.keep
      remove  config/initializers/cors.rb
      remove  config/initializers/new_framework_defaults_8_1.rb
         run  bundle install --quiet
         run  bundle lock --add-platform=x86_64-linux
Writing lockfile to /Users/kerrick/Temporary/sdoc_test/Gemfile.lock
         run  bundle lock --add-platform=aarch64-linux
Writing lockfile to /Users/kerrick/Temporary/sdoc_test/Gemfile.lock
       rails  importmap:install
       apply  /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/importmap-rails-2.2.2/lib/install/install.rb
  Add Importmap include tags in application layout
      insert    app/views/layouts/application.html.erb
  Create application.js module as entrypoint
      create    app/javascript/application.js
  Use vendor/javascript for downloaded pins
      create    vendor/javascript
      create    vendor/javascript/.keep
  Configure importmap paths in config/importmap.rb
      create    config/importmap.rb
  Copying binstub
      create    bin/importmap
         run  bundle install --quiet
       rails  turbo:install stimulus:install
       apply  /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/turbo-rails-2.0.20/lib/install/turbo_with_importmap.rb
  Import Turbo
      append    app/javascript/application.js
  Pin Turbo
      append    config/importmap.rb
         run  bundle install --quiet
       apply  /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/stimulus-rails-1.3.4/lib/install/stimulus_with_importmap.rb
  Create controllers directory
      create    app/javascript/controllers
      create    app/javascript/controllers/index.js
      create    app/javascript/controllers/application.js
      create    app/javascript/controllers/hello_controller.js
  Import Stimulus controllers
      append    app/javascript/application.js
  Pin Stimulus
  Appending: pin "@hotwired/stimulus", to: "stimulus.min.js"
      append    config/importmap.rb
  Appending: pin "@hotwired/stimulus-loading", to: "stimulus-loading.js"
      append    config/importmap.rb
  Pin all controllers
  Appending: pin_all_from "app/javascript/controllers", under: "controllers"
      append    config/importmap.rb
         run  bundle install --quiet
         run  bundle binstubs kamal
         run  bundle exec kamal init
Created configuration file in config/deploy.yml
Created .kamal/secrets file
Created sample hooks in .kamal/hooks
       force  .kamal/secrets
       force  config/deploy.yml
       rails  solid_cache:install solid_queue:install solid_cable:install
      create  config/cache.yml
      create  db/cache_schema.rb
        gsub  config/environments/production.rb
      create  config/queue.yml
      create  config/recurring.yml
      create  db/queue_schema.rb
      create  bin/jobs
        gsub  config/environments/production.rb
      create  db/cable_schema.rb
       force  config/cable.yml
kerrick@laptop-kerrick-primary Temporary % cd sdoc_test 
kerrick@laptop-kerrick-primary sdoc_test % cat >> Gemfile
  gem "sdoc", require: false, git: "https://github.com/zzak/sdoc.git", branch: "re-379", group: :development
kerrick@laptop-kerrick-primary sdoc_test % bundle
Fetching https://github.com/zzak/sdoc.git
Fetching gem metadata from https://rubygems.org/.........
Resolving dependencies...
Bundle complete! 24 Gemfile dependencies, 126 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
kerrick@laptop-kerrick-primary sdoc_test % sdoc .   
zsh: command not found: sdoc
kerrick@laptop-kerrick-primary sdoc_test % bundle exec sdoc .
Parsing sources...
100% [1431/1431]  tmp/cache/bootsnap/load-path-cache

Generating SDoc format into /Users/kerrick/Temporary/sdoc_test/doc...

You can visit the home page at: file:///Users/kerrick/Temporary/sdoc_test/doc/index.html
uh-oh! RDoc had a problem:
Error while evaluating /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/rdoc/generator/template/sdoc/_context.rhtml: undefined method 'each_constant' for an instance of RDoc::NormalClass (at "\n      <table border='0' cellpadding='5'>\n        ")

run with --debug for full backtrace
kerrick@laptop-kerrick-primary sdoc_test % bundle exec sdoc . --debug
Parsing sources...
100% [1431/1431]  tmp/cache/bootsnap/load-path-cache

Generating SDoc format into /Users/kerrick/Temporary/sdoc_test/doc...

You can visit the home page at: file:///Users/kerrick/Temporary/sdoc_test/doc/index.html
Copying /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/rdoc/generator/template/sdoc/resources/./** to /Users/kerrick/Temporary/sdoc_test/doc/**
Generating JSON index
  writing search index to js/search_index.js
  generating class search index
    ApplicationController
    ApplicationHelper
    ApplicationJob
    ApplicationMailer
    ApplicationRecord
    Object
    SdocTest
    SdocTest::Application
  generating method search index
    Object#system!
  generating pages search index
    Dockerfile
    Gemfile
    Gemfile.lock
    README
    Rakefile
    application.css
    application.js
    application.js
    hello_controller.js
    index.js
    service-worker.js
    docker-entrypoint
    config.ru
    credentials.yml.enc
    master.key
    development.log
    400.html
    404.html
    406-unsupported-browser.html
    422.html
    500.html
    robots
mkdir -p /Users/kerrick/Temporary/sdoc_test/doc/js
install -c -p -m 644 js/navigation.js /Users/kerrick/Temporary/sdoc_test/doc/js/navigation.js
install -c -p -m 644 js/searcher.js /Users/kerrick/Temporary/sdoc_test/doc/js/searcher.js
Compressing generated JSON index
Reading the JSON index file from /Users/kerrick/Temporary/sdoc_test/doc/js/search_index.js
Writing gzipped search index to /Users/kerrick/Temporary/sdoc_test/doc/js/search_index.js.gz
Reading the original js file from /Users/kerrick/Temporary/sdoc_test/doc/js/navigation.js
Writing gzipped file to /Users/kerrick/Temporary/sdoc_test/doc/js/navigation.js.gz
Reading the original js file from /Users/kerrick/Temporary/sdoc_test/doc/js/searcher.js
Writing gzipped file to /Users/kerrick/Temporary/sdoc_test/doc/js/searcher.js.gz
Generating search engine index in /Users/kerrick/Temporary/sdoc_test/doc
Generating class tree
  writing class tree to panel/tree.js
Generating index file in /Users/kerrick/Temporary/sdoc_test/doc
Generating file documentation in /Users/kerrick/Temporary/sdoc_test/doc
  working on Dockerfile (/Users/kerrick/Temporary/sdoc_test/doc/files/Dockerfile.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/Dockerfile.html
  working on Gemfile (/Users/kerrick/Temporary/sdoc_test/doc/files/Gemfile.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/Gemfile.html
  working on Gemfile.lock (/Users/kerrick/Temporary/sdoc_test/doc/files/Gemfile_lock.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/Gemfile_lock.html
  working on README.md (/Users/kerrick/Temporary/sdoc_test/doc/files/README_md.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/README_md.html
  working on Rakefile (/Users/kerrick/Temporary/sdoc_test/doc/files/Rakefile.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/Rakefile.html
  working on app/assets/stylesheets/application.css (/Users/kerrick/Temporary/sdoc_test/doc/files/app/assets/stylesheets/application_css.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/assets/stylesheets/application_css.html
  working on app/controllers/application_controller.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/app/controllers/application_controller_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/controllers/application_controller_rb.html
  working on app/helpers/application_helper.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/app/helpers/application_helper_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/helpers/application_helper_rb.html
  working on app/javascript/application.js (/Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/application_js.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/application_js.html
  working on app/javascript/controllers/application.js (/Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/controllers/application_js.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/controllers/application_js.html
  working on app/javascript/controllers/hello_controller.js (/Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/controllers/hello_controller_js.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/controllers/hello_controller_js.html
  working on app/javascript/controllers/index.js (/Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/controllers/index_js.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/javascript/controllers/index_js.html
  working on app/jobs/application_job.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/app/jobs/application_job_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/jobs/application_job_rb.html
  working on app/mailers/application_mailer.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/app/mailers/application_mailer_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/mailers/application_mailer_rb.html
  working on app/models/application_record.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/app/models/application_record_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/models/application_record_rb.html
  working on app/views/pwa/service-worker.js (/Users/kerrick/Temporary/sdoc_test/doc/files/app/views/pwa/service-worker_js.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/app/views/pwa/service-worker_js.html
  working on bin/brakeman (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/brakeman.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/brakeman.html
  working on bin/bundler-audit (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/bundler-audit.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/bundler-audit.html
  working on bin/ci (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/ci.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/ci.html
  working on bin/dev (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/dev.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/dev.html
  working on bin/docker-entrypoint (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/docker-entrypoint.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/docker-entrypoint.html
  working on bin/importmap (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/importmap.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/importmap.html
  working on bin/jobs (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/jobs.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/jobs.html
  working on bin/kamal (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/kamal.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/kamal.html
  working on bin/rails (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/rails.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/rails.html
  working on bin/rake (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/rake.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/rake.html
  working on bin/rubocop (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/rubocop.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/rubocop.html
  working on bin/setup (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/setup.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/setup.html
  working on bin/thrust (/Users/kerrick/Temporary/sdoc_test/doc/files/bin/thrust.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/bin/thrust.html
  working on config.ru (/Users/kerrick/Temporary/sdoc_test/doc/files/config_ru.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config_ru.html
  working on config/application.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/application_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/application_rb.html
  working on config/boot.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/boot_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/boot_rb.html
  working on config/ci.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/ci_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/ci_rb.html
  working on config/credentials.yml.enc (/Users/kerrick/Temporary/sdoc_test/doc/files/config/credentials_yml_enc.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/credentials_yml_enc.html
  working on config/environment.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/environment_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/environment_rb.html
  working on config/environments/development.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/environments/development_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/environments/development_rb.html
  working on config/environments/production.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/environments/production_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/environments/production_rb.html
  working on config/environments/test.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/environments/test_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/environments/test_rb.html
  working on config/importmap.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/importmap_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/importmap_rb.html
  working on config/initializers/assets.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/assets_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/assets_rb.html
  working on config/initializers/content_security_policy.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/content_security_policy_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/content_security_policy_rb.html
  working on config/initializers/filter_parameter_logging.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/filter_parameter_logging_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/filter_parameter_logging_rb.html
  working on config/initializers/inflections.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/inflections_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/initializers/inflections_rb.html
  working on config/master.key (/Users/kerrick/Temporary/sdoc_test/doc/files/config/master_key.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/master_key.html
  working on config/puma.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/puma_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/puma_rb.html
  working on config/routes.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/config/routes_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/config/routes_rb.html
  working on db/cable_schema.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/db/cable_schema_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/db/cable_schema_rb.html
  working on db/cache_schema.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/db/cache_schema_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/db/cache_schema_rb.html
  working on db/queue_schema.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/db/queue_schema_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/db/queue_schema_rb.html
  working on db/seeds.rb (/Users/kerrick/Temporary/sdoc_test/doc/files/db/seeds_rb.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/db/seeds_rb.html
  working on log/development.log (/Users/kerrick/Temporary/sdoc_test/doc/files/log/development_log.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/log/development_log.html
  working on public/400.html (/Users/kerrick/Temporary/sdoc_test/doc/files/public/400_html.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/public/400_html.html
  working on public/404.html (/Users/kerrick/Temporary/sdoc_test/doc/files/public/404_html.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/public/404_html.html
  working on public/406-unsupported-browser.html (/Users/kerrick/Temporary/sdoc_test/doc/files/public/406-unsupported-browser_html.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/public/406-unsupported-browser_html.html
  working on public/422.html (/Users/kerrick/Temporary/sdoc_test/doc/files/public/422_html.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/public/422_html.html
  working on public/500.html (/Users/kerrick/Temporary/sdoc_test/doc/files/public/500_html.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/public/500_html.html
  working on public/robots.txt (/Users/kerrick/Temporary/sdoc_test/doc/files/public/robots_txt.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/files/public/robots_txt.html
Generating class documentation in /Users/kerrick/Temporary/sdoc_test/doc
  working on ApplicationController (classes/ApplicationController.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/classes/ApplicationController.html
  working on ApplicationHelper (classes/ApplicationHelper.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/classes/ApplicationHelper.html
  working on ApplicationJob (classes/ApplicationJob.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/classes/ApplicationJob.html
  working on ApplicationMailer (classes/ApplicationMailer.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/classes/ApplicationMailer.html
  working on ApplicationRecord (classes/ApplicationRecord.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/classes/ApplicationRecord.html
  working on Object (classes/Object.html)
  rendering /Users/kerrick/Temporary/sdoc_test/doc/classes/Object.html
Error while evaluating /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/rdoc/generator/template/sdoc/_context.rhtml: undefined method 'each_constant' for an instance of RDoc::NormalClass (at "\n      <table border='0' cellpadding='5'>\n        ")
/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/rdoc/generator/template/sdoc/_context.rhtml:106:in 'block in SDoc::Templatable#include_template'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/rdoc-6.15.1/lib/rdoc/code_object/context.rb:745:in 'block in RDoc::Context#each_section'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/rdoc-6.15.1/lib/rdoc/code_object/context.rb:744:in 'Array#each'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/rdoc-6.15.1/lib/rdoc/code_object/context.rb:744:in 'RDoc::Context#each_section'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/rdoc/generator/template/sdoc/_context.rhtml:88:in 'SDoc::Templatable#include_template'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/erb-6.0.0/lib/erb.rb:1012:in 'Kernel#eval'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/erb-6.0.0/lib/erb.rb:1012:in 'ERB#result'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/templatable.rb:17:in 'SDoc::Templatable#eval_template'
	(eval at /Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/templatable.rb:32):1:in 'SDoc::Templatable#include_template'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/templatable.rb:32:in 'Kernel#eval'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/templatable.rb:32:in 'SDoc::Templatable#include_template'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/rdoc/generator/template/sdoc/class.rhtml:48:in 'block in RDoc::Generator::SDoc#generate_class_files'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/erb-6.0.0/lib/erb.rb:1012:in 'Kernel#eval'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/erb-6.0.0/lib/erb.rb:1012:in 'ERB#result'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/templatable.rb:17:in 'SDoc::Templatable#eval_template'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/templatable.rb:39:in 'SDoc::Templatable#render_template'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/generator.rb:147:in 'block in RDoc::Generator::SDoc#generate_class_files'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/generator.rb:141:in 'Array#each'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/generator.rb:141:in 'RDoc::Generator::SDoc#generate_class_files'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/lib/sdoc/generator.rb:106:in 'RDoc::Generator::SDoc#generate'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/rdoc-6.15.1/lib/rdoc/rdoc.rb:513:in 'block in RDoc::RDoc#generate'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/rdoc-6.15.1/lib/rdoc/rdoc.rb:507:in 'Dir.chdir'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/rdoc-6.15.1/lib/rdoc/rdoc.rb:507:in 'RDoc::RDoc#generate'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/rdoc-6.15.1/lib/rdoc/rdoc.rb:486:in 'RDoc::RDoc#document'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bundler/gems/sdoc-3908b20a3275/bin/sdoc:7:in '<top (required)>'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bin/sdoc:25:in 'Kernel#load'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/bin/sdoc:25:in '<top (required)>'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Kernel.load'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Bundler::CLI::Exec#kernel_load'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/cli/exec.rb:23:in 'Bundler::CLI::Exec#run'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/cli.rb:452:in 'Bundler::CLI#exec'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/command.rb:28:in 'Bundler::Thor::Command#run'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in 'Bundler::Thor::Invocation#invoke_command'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor.rb:538:in 'Bundler::Thor.dispatch'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/cli.rb:35:in 'Bundler::CLI.dispatch'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/base.rb:584:in 'Bundler::Thor::Base::ClassMethods#start'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/cli.rb:29:in 'Bundler::CLI.start'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/bundler-2.6.9/exe/bundle:28:in 'block in <top (required)>'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/3.4.0/bundler/friendly_errors.rb:117:in 'Bundler.with_friendly_errors'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/lib/ruby/gems/3.4.0/gems/bundler-2.6.9/exe/bundle:20:in '<top (required)>'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/bin/bundle:25:in 'Kernel#load'
	/Users/kerrick/.local/share/mise/installs/ruby/3.4.7/bin/bundle:25:in '<main>'

kerrick@laptop-kerrick-primary sdoc_test % 
```

</details>

zzak committed:
Use constants variable [sdoc template]

This code `context.each_section do |section, constants, attributes|`
provides the list of contents for us.

Originally fixed for [rails template] in f691448.

Co-authored-by: Code Ass <aycabta@gmail.com>

zzak commented:
@Kerrick Thanks for testing it out and reporting back!

I found this was fixed in rails/sdoc@f691448 for the `rails` template, so just copied it over. Should be working now, give it a try and let me know if there are any more issues before we try to release this.

optimuspwnius commented:
I've been using this pr for awhile, works great, no issues! Looking forward to a release. 👍
